subsys/bluetooth/host/scan.c

```diff
@@ -1335,6 +1335,15 @@ void bt_le_scan_cb_unregister(struct bt_le_scan_cb *cb)
 	sys_slist_find_and_remove(&scan_cbs, &cb->node);
 }
 
+void bt_le_scan_uninit(void)
+{
+	struct bt_le_scan_cb *cb;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&scan_cbs, cb, node) {
+		sys_slist_find_and_remove(&scan_cbs, &cb->node);
+	}
```

Comment:
Consider using something like this to avoid iterating over the list unnecessarily:
```suggestion
	while (sys_slist_get(&scan_cbs)) {}
```

Reply:
Forget the thing about extra work. Both versions will remove the first element in a loop.

However, the [docs](https://docs.zephyrproject.org/latest/kernel/data_structures/slist.html#c.SYS_SLIST_FOR_EACH_CONTAINER) for `SYS_SLIST_FOR_EACH_CONTAINER` states:
> Note: the loop is unsafe and thus __cn should not be detached.

Reply:
What does "should not be detached" mean, sorry? So do you reckon `while (sys_slist_get(&scan_cbs)) {}` is the better way to go?

Reply:
I interpret it as "should not be removed from the list". `while (sys_slist_get(&scan_cbs)) {}` is more in line with how we do it for buffers. For the list we could also set the `head` and `tail` pointer to `NULL` to "forget" the list, but then the rest of the list would still exist (the nodes would point to the next nodes still). I am not sure if it matters.